### PR TITLE
feat(elements): web custom element generator

### PR DESCRIPTION
**Pull request by NathanWalker**
* [x] Generate custom web elements from any shared code in workspace
* [x] Provide build configuration to wrap the custom elements into a single file for insertion into any web page
* [x] Provide preview app which allows the build to insert the compiled element build to preview your custom elements working

Example of usage:
```
ng g elements ui-kit --barrel=@mycompany/ui --components=menu,footer
```

Doing so will generate an elements module in `xplat/web/elements/ui-kit.module.ts`:
*NOTE:* The component prefix selector for each custom element will be whatever prefix your workspace is set to use.
```
import { NgModule, Injector } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { createCustomElement } from '@angular/elements';
import { MenuComponent, FooterComponent } from '@mycompany/web';

@NgModule({
  imports: [BrowserModule],
  entryComponents: [
    MenuComponent, FooterComponent
  ]
})
export class UiKitModule {
  constructor(private injector: Injector) {
    
  }

  ngDoBootstrap() {
    let component;
    component = createCustomElement(MenuComponent, { injector: this.injector });
    customElements.define('abc-menu', component);
    component = createCustomElement(FooterComponent, { injector: this.injector });
    customElements.define('abc-footer', component);
  }
}
```

This module will then be able to be imported into the build configurator one that's complete here.

### CODE REVIEW

File: src/elements/_files/__name__.module.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { NgModule, Injector } from '@angular/core';
+import { createCustomElement } from '@angular/elements';
+import { <%= componentSymbolList %> } from '<%= barrel %>';
+
+@NgModule({
+  imports: [],
+  exports: [],
+  declarations: [],
+  entryComponents: [
+    <%= componentSymbolList %>
```
Comment (JoshDSommer) [inline src/elements/_files/__name__.module.ts]: Should this be a list? I'm not 100% sure about this but I was thinking that it was only possible to export a single component at a time as an Angular Element? I could be completely wrong about this and made the wrong assumption though, but figured I would mention it to be sure.

File: src/elements/index.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+function updateWorkspaceSupport(options: ElementsOptions, tree: Tree, context: SchematicContext) {
+  return updateJsonInTree("package.json", json => {
+    json.scripts = json.scripts || {};
+    json.dependencies = json.dependencies || {};
+    const angularVersion = json.dependencies['@angular/core'];
+    json.dependencies = {
+      ...json.dependencies,
+      "@angular/elements": angularVersion,
+      "@webcomponents/webcomponentsjs": "^2.2.7",
+      "http-server": "^0.11.1",
```
Comment (JoshDSommer) [inline src/elements/index.ts]: We shouldn't need http server unless there are providing a testing index.html page ground for them, to load the exported elements in. That does seem worth while though, maybe as a e2e test or something in a ci/cd process.

File: src/elements/_files/__name__.module.ts (hunk 1 above)
Comment (NathanWalker) [reply src/elements/_files/__name__.module.ts]: List works well 🎉

File: src/elements/index.ts (hunk 2 above)
Comment (NathanWalker) [reply src/elements/index.ts]: I'll be adding ability to quick preview your elements after you generate them to verify they work with this 👍

**Pull request merged**